lib/rucio/daemons/rsedecommissioner/profiles/generic.py

```diff
@@ -0,0 +1,477 @@
+    - The RSE expression of the rule is trivial (the RSE name itself).
+    - There are no replicas locked by the rule that reside on another RSE.
+
+    :param rse: RSE table entry as a dictionary.
+    :param config: Decommissioning configuration dictionary.
+    :param logger: Logging function.
+    """
+    handlers = [
+        (_is_being_deleted, _count_as_processed),
+        (_has_trivial_rse_expression, _delete_rule),
+        (_has_last_replica_on_rse, _delete_rule),
+    ]
+    generic_profile_template_function(rse, config,
+                                      _generic_initialize,
+                                      list_rules_for_rse_decommissioning,
+                                      handlers,
+                                      _generic_finalize,
+                                      logger=logger)
+
+
+def generic_move(
+    rse: 'dict[str, Any]',
+    config: 'dict[str, Any]',
+    *,
+    logger: 'Callable[..., None]' = logging.log
+) -> None:
+    """Move rules that satisfy conditions to a specific destination.
+
+    The "generic move" profile lists out all rules that are locking replicas at the given RSE,
+    and moves them to the specified destination if either one of the following is true:
+
+    - The RSE expression of the rule is trivial (the RSE name itself).
+    - There are no replicas locked by the rule that reside on another RSE.
+
+    :param rse: RSE table entry as a dictionary.
+    :param config: Decommissioning configuration dictionary.
+    :param logger: Logging function.
+    """
+    move_to_destination = RuleMover(config['destination'])
+    handlers = [
+        (_is_being_deleted, _count_as_processed),
+        (_has_child_rule_id, _count_as_processed),
+        (_has_trivial_rse_expression, move_to_destination),
+        (_has_last_replica_on_rse, move_to_destination),
+    ]
+    generic_profile_template_function(rse, config,
+                                      _generic_initialize,
+                                      list_rules_for_rse_decommissioning,
+                                      handlers,
+                                      _generic_finalize,
+                                      logger=logger)
+
+
+def _generic_initialize(
+    rse: 'dict[str, Any]',
+    status: 'DecommissioningStatus',
+    *,
+    logger: 'Callable[..., None]' = logging.log,
+) -> None:
+    """Set the RSE availability flags and the decommissioning status.
+
+    When an RSE is initialized for decommissioning, the availability flags are set as follows:
+
+    - `availability_read=True`
+    - `availability_write=False`
+    - `availability_delete=True`
+
+    and the following attribute is set:
+
+    - `greedyDeletion=True`
+
+    :param rse: RSE table entry as a dictionary.
+    :param status: Decommissioning status.
+    :param logger: Logging function.
+    """
+
+    if status != DecommissioningStatus.PROCESSING:
+        return
+
+    # Set the RSE availability and greedyDeletion attributes.
```

Comment:
Given the log entry below, I believe this comment is not useful.

Reply:
Removed the comment.